extensions/applicationinsights-dependencies-js/src/ajaxRecord.ts

```diff
@@ -1,83 +1,351 @@
+    *  +------------+-+---+----------------+-+--+--------+-+-----------+-+------------+-+
+    *  |--redirect--| |---|--domainLookup--| |--connect--| |--request--| |--response--| |
+    *  |-------------------networkConnect----------------|
+    *  |                                                   |---------sentRequest--------|
+    *  |------------------------------------perfTotal-----------------------------------|
+    */
+
+    let resourceEntry = ajaxData.perfTiming;
+    let props = dependency[strProperties] || {};
+    let propsSet = 0;
+
+    if (resourceEntry) {
+        // redirect
+        propsSet |= _setPerfDuration(props, strRedirect, resourceEntry, strRedirectStart, strRedirectEnd);
+
+        // domainLookup
+        propsSet |= _setPerfDuration(props, strDomainLookup, resourceEntry, strDomainLookupStart, strDomainLookupEnd);
+
+        // connect
+        propsSet |= _setPerfDuration(props, strConnect, resourceEntry, strConnectStart, strConnectEnd);
+
+        // request
+        propsSet |= _setPerfDuration(props, strRequest, resourceEntry, strRequestStart, strRequestEnd);
+
+        // response
+        propsSet |= _setPerfDuration(props, strResponse, resourceEntry, strResponseStart, strResponseEnd);
+
+        // Network connection time
+        propsSet |= _setPerfDuration(props, "networkConnect", resourceEntry, strStartTime, strConnectEnd);
+
+        // Sent Request
+        propsSet |= _setPerfDuration(props, "sentRequest", resourceEntry, strRequestStart, strResponseEnd);
+
+        // PerfTotal / Duration
+        let duration = resourceEntry[strDuration];
+        if (!duration) {
+            duration = _calcPerfDuration(resourceEntry, strStartTime, strResponseEnd)
+        }
+        if (duration) {
+            propsSet |= _setPerfValue(props, strDuration, duration);
+            propsSet |= _setPerfValue(props, "perfTotal", duration);
+        }
+
+        var serverTiming = resourceEntry[strServerTiming];
+        if (serverTiming) {
+            let server = [];
+            _arrForEach(serverTiming, (value) => {
+                let newValue = {};
+                _arrForEach(_objKeys(value), (key) => {
+                    newValue[key] = value[key];
+                });
+                server.push(newValue);
+            });
+            propsSet |= _setPerfValue(props, strServerTiming, server);
+        }
+
+        propsSet |= _setPerfValue(props, strTransferSize, resourceEntry[strTransferSize])
+        propsSet |= _setPerfValue(props, strEncodedBodySize, resourceEntry[strEncodedBodySize])
+        propsSet |= _setPerfValue(props, strDecodedBodySize, resourceEntry[strDecodedBodySize])
+    } else {
+        if (ajaxData.perfMark) {
+            propsSet |= _setPerfValue(props, "missing", ajaxData.perfAttempts);
+        }
+    }
+
+    if (propsSet) {
+        dependency[strProperties] = props;
+    }
+}
 
 export class XHRMonitoringState {
-    public openDone: boolean = false;
```

Comment:
As eariler, moving populating the values so that Typescript doesn't place this in the constructor as "this.xxxx" and instead we can use self.xxxx